### PR TITLE
feat: Refactor Frondend with Admin API

### PR DESCRIPTION
**Pull request by juzhiyuan**
- [x] Upstream
- [x] Route
- [ ] SSL
- [x] Consumer
- [x] Route: [proxy-rewrite](https://github.com/apache/apisix/blob/master/doc/plugins/proxy-rewrite.md) Headers
- [ ] Status Code
- [x] Pagination

close https://github.com/apache/apisix-dashboard/issues/498

### CODE REVIEW

File: src/pages/Route/transform.ts [hunk 1]
```diff
@@ -77,6 +77,8 @@ export const transformStepData = ({
       'redirectURI',
       'ret_code',
       'redirectOption',
+      !Object.keys(step3Data.plugins).length ? 'plugins' : '',
```
Comment (juzhiyuan) [inline src/pages/Route/transform.ts:80]: Why we need `!`? @LiteSun

File: src/pages/Route/components/Step2/RequestRewriteView.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ const RequestRewriteView: React.FC<RouteModule.Step2PassProps> = ({
     fetchUpstreamList().then(({ data }) => setList(data));
   }, []);
   return (
-    <UpstreamForm ref={upstreamRef} form={form} disabled={disabled} list={list} showSelector />
+    <UpstreamForm ref={upstreamRef} form={form} disabled={disabled} list={list} showSelector key={1} />
```
Comment (juzhiyuan) [inline src/pages/Route/components/Step2/RequestRewriteView.tsx:32]: Why we need this key?

Comment (LiteSun) [reply src/pages/Route/components/Step2/RequestRewriteView.tsx:32]: this key uses for fix route step2 warnings.

File: src/pages/Route/components/Step1/RequestConfigView.tsx [hunk 3]
```diff
@@ -99,8 +99,8 @@ const RequestConfigView: React.FC<RouteModule.Step1PassProps> = ({
     </Form.List>
   );
 
-  const renderPaths = () => (
-    <Form.List name="paths">
+  const renderUris = () => (
```
Comment (juzhiyuan) [inline src/pages/Route/components/Step1/RequestConfigView.tsx:102]: <URIList />

Comment (juzhiyuan) [reply src/pages/Route/components/Step1/RequestConfigView.tsx:102]: <URIList />

**Comment by juzhiyuan** [comment]
@moonming Please check if there have any License issues.

@liuxiran @bzp2010 Please take a looks at those code changes.

**Review by LiteSun — APPROVED** [review]

**Comment by juzhiyuan** [comment]
cc @moonming @liuxiran This PR is going to merge into the master branch, and it depends on the api refactor branch, so it means the master branch is unstable after this PR is merged, @nic-chen @ShiningRush please Pull Request the refactor branch once it’s ready.

**Comment by moonming** [comment]
got it
Thanks,
Ming Wen
Twitter: _WenMing


琚致远 <notifications@github.com> 于2020年10月19日周一 下午4:15写道：

**Comment by juzhiyuan** [comment]
There would have a lot conflict changes, once the refactor branch is merged into master, we will handle those conflicts

**Comment by juzhiyuan** [comment]
ping @moonming @liuxiran

**Review by moonming — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Pull request merged**